templates/cluster-template-external-cloud-provider.yaml

```diff
@@ -0,0 +1,158 @@
+---
+apiVersion: cluster.x-k8s.io/v1alpha3
+kind: Cluster
+metadata:
+  name: ${CLUSTER_NAME}
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks: ["192.168.0.0/16"] # CIDR block used by Calico.
+    serviceDomain: "cluster.local"
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+    kind: OpenStackCluster
+    name: ${CLUSTER_NAME}
+  controlPlaneRef:
+    kind: KubeadmControlPlane
+    apiVersion: controlplane.cluster.x-k8s.io/v1alpha3
+    name: ${CLUSTER_NAME}-control-plane
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+kind: OpenStackCluster
+metadata:
+  name: ${CLUSTER_NAME}
+spec:
+  cloudName: ${OPENSTACK_CLOUD}
+  cloudsSecret:
+    name: ${CLUSTER_NAME}-cloud-config
+    namespace: ${NAMESPACE}
+  managedAPIServerLoadBalancer: true
+  apiServerLoadBalancerFloatingIP: ${OPENSTACK_CONTROLPLANE_IP}
+  apiServerLoadBalancerPort: 6443
+  managedSecurityGroups: true
+  nodeCidr: 10.6.0.0/24
+  dnsNameservers:
+  - ${OPENSTACK_DNS_NAMESERVERS}
+  externalNetworkId: ${OPENSTACK_EXTERNAL_NETWORK_ID}
+  disablePortSecurity: false
+  disableServerTags: true
+  useOctavia: true
+---
```

Comment:
I believe the only difference is the kubeadm template for master and worker between internal/external?
so later on we may consider how to change it to avoid maintain too many templates

Reply:
Partially Yes. The difference between internal and external for creating cluster is only the template. But there is a huge difference after creating cluster. If we use external cloud provider, we have to create a secret and apply three yamls to create openstack-cloud-controller-manager.

Reply:
oh... understand your point now,maybe we can keep this one and overtime the in tree provider will be removed
so how about put all those things into `outtree` folder and put original one into `intree` folder?

Reply:
As far as seeing docs like [Cloud Providers](https://github.com/kubernetes/kubernetes/tree/master/pkg/cloudprovider) and [Deprecation Notice](https://github.com/kubernetes/kubernetes/tree/master/pkg/cloudprovider), `external` is common these days.
I think `external` is more operator's point of view and better.

Reply:
so you mean by default to out of tree? I am ok to that and I think we need keep intree for a while (reason such as stable, someone might use it )

Reply:
We do not change directory structure. This means default is intree because `cluster-template.yaml` is used by default.

Reply:
both fine to me ..